cfe_ts_crc.c

```diff
@@ -134,7 +134,12 @@ int main(int argc, char **argv)
     while (done == 0)
     {
         readSize = read(fd, buffer, 100);
-        fileCRC  = CalculateCRC(buffer, readSize, fileCRC);
+        if (readSize < 0)
+        {
+            printf("\ncfe_ts_crc error: file read failed!\n");
+            exit(1);
+        }
+        fileCRC = CalculateCRC(buffer, (uint32)readSize, fileCRC);
```

Comment:
It was done because CalculateCRC is expecting a uint32

Reply:
But it will convert implicitly - and forcing a conversion can have unintended consequences.  In reality the CalculateCRC() functions should take a `size_t` anyway.

Note that C has very specific rules about when and how implicit type conversions are done - and things (almost) always just work.  When it doesn't just work you'll get a compiler warning about it, but a cast can hide that warning because it is literally telling the compiler "I meant to do that".  Hence why its not advised to put them in unless there is a real reason why you need to tell the compiler that you meant to do that.  In this case, I don't see any reason.

Reply:
Note - I'm not suggesting changing CalculateCRC to `size_t` here - just leave it alone - it doesn't matter.  My only concern is the extra cast - should not do this.